File: src/Client/Logistics.OfficeApp/src/app/core/enums/loadStatus.ts
import {EnumLike, getEnumDescription} from './enumLike';

export enum LoadStatus {
  Dispatched,
  PickedUp,
  Delivered
}

export const LoadStatusEnum: EnumLike = {
  Dispatched: {value: 0, description: 'Dispatched'},
  PickedUp: {value: 1, description: 'Picked Up'},
  Delivered: {value: 2, description: 'Delivered'},

  getDescription(value: string | number): string {
    return getEnumDescription(this, value);
  },
};

File: src/Shared/Logistics.Shared.Consts/Enums/Load/LoadStatus.cs
using System.ComponentModel;
using System.Runtime.Serialization;

namespace Logistics.Shared.Consts;

public enum LoadStatus
{
    [Description("Dispatched"), EnumMember(Value = "dispatched")] 
    Dispatched,
    
    [Description("Picked Up"), EnumMember(Value = "picked_up")] 
    PickedUp,
    
    [Description("Delivered"), EnumMember(Value = "delivered")] 
    Delivered
}
